sbt-test/compilerReporter/i14576/Test.scala

```diff
@@ -13,5 +13,4 @@ object Test:
   // private[this] and = _ are deprecated under -source:future
   private[this] var x: AnyRef = _
 
-  // under -source:future, `_` is deprecated for wildcard arguments of types: use `?` instead
   val xs: List[_] = Nil
```

Comment:
This change does not seem right. In that position, `List[_]` is not a pattern type, so `_` should stay deprecated.

Reply:
Thanks very much, I've removed the change to the test now. The fix isn't as trivial as I thought it was.

Reply:
Currently this warining is not properly tested for cases that are not in patterns. I am adding tests in  #18813.